### PR TITLE
Adding Hard Sigmoid Activation

### PR DESCRIPTION
**Pull request by abhinavarora**
Fixes https://github.com/PaddlePaddle/Paddle/issues/4605

### CODE REVIEW

File: paddle/operators/activation_op.cc [hunk 1]
```diff
@@ -338,6 +338,37 @@ class ThresholdedReluOpMaker : public framework::OpProtoAndCheckerMaker {
...
+    AddInput("X", "Input of HardSigmoid operator");
+    AddOutput("Y", "Output of HardSigmoid operator");
+    AddComment(R"DOC(
+Hard Sigmoid activation operator.
+
+Segment-wise linear approximation of sigmoid[1].
+This is much faster than sigmoid.
+
+hard_sigmoid = max(0, min(1, slope * x + shift))
+
```
Comment (sidgoyal78) [inline paddle/operators/activation_op.cc]: Is it true that both slope and shift are non-negative? If yes, it will be helpful to mention as a comment.

Comment (abhinavarora) [reply paddle/operators/activation_op.cc]: Fixed in f4820c7dd10f54eb3772a4abc5b3f408072cbbae

File: paddle/operators/activation_op.cc [hunk 2]
```diff
@@ -338,6 +338,38 @@ class ThresholdedReluOpMaker : public framework::OpProtoAndCheckerMaker {
...
 };
 
+template <typename AttrType>
+class HardSigmoidOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  HardSigmoidOpMaker(framework::OpProto *proto,
+                     framework::OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "Input of HardSigmoid operator");
+    AddOutput("Y", "Output of HardSigmoid operator");
```
Comment (lcy-seso) [inline paddle/operators/activation_op.cc]: Out [name convention documentation](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/name_convention.md) suggest us use the "(type, default value) usage" style to write the comments. I guess this comment can be further refined. And fc_op is a good example: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/fc_op.cc#L122.

Comment (abhinavarora) [reply paddle/operators/activation_op.cc]: @lcy-seso  Do you recommend that I modify this for all ops in the activation.cc? Unfortunately all activations in this file do not follow the naming convention. The other option is that I merge this PR and open an issue to fix the naming convention of all Ops in Activations? The second option will keep this PR clean. What do you suggest?

File: paddle/operators/activation_op.cc (hunk 1 above)
Comment (abhinavarora) [reply paddle/operators/activation_op.cc]: Corrected comment grammatical mistake in 7b1054676248a12ec536d36a0ffab34b29c0798b

File: paddle/operators/activation_op.cc (hunk 2 above)
Comment (lcy-seso) [reply paddle/operators/activation_op.cc]: I prefer we guarantee all the newly written unmerged PR follows this naming convention first, and make a new PR later to fix all the old codes.

Comment (abhinavarora) [reply paddle/operators/activation_op.cc]: Thank you for pointing this out. Opened a separate issue to fix this. https://github.com/PaddlePaddle/Paddle/issues/4772

**Pull request merged**